### PR TITLE
feat(types): add typescript 4.8 compatibility

### PR DESCRIPTION
**Pull request by bnorbi95**
### Pull Request Checklist

_Please make sure to review and check all of these items:_

- [ ] Have you added new tests to prevent regressions?
- [ ] Does `yarn test` or `yarn test-DIALECT` pass with this change (including linting)?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you update the typescript typings accordingly (if applicable)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)?

### Description Of Change

Sequelize typings are affected in a few places by a TypeScript 4.8 breaking change (https://devblogs.microsoft.com/typescript/announcing-typescript-4-8/#unconstrained-generics-no-longer-assignable-to).

This PR addresses #14934.

### CODE REVIEW

**Review by fzn0x — COMMENTED** [review]
Can you write types tests for this?

**Comment by WikiRik** [comment]
Thanks for the PR!
I've gone ahead and added the new TypeScript versions to the CI in this PR. I hope you don't mind.

Are you willing to make a PR towards the main branch as well?

File: src/hooks.d.ts [hunk 1]
```diff
@@ -67,8 +67,8 @@ export interface ModelHooks<M extends Model = Model, TAttributes = any> {
 
 export interface SequelizeHooks<
   M extends Model<TAttributes, TCreationAttributes> = Model,
-  TAttributes = any,
-  TCreationAttributes = TAttributes
+  TAttributes extends {} = any,
+  TCreationAttributes extends {}= TAttributes
```
Comment (ephys) [inline src/hooks.d.ts:71]: Tiny spacing issue here

```suggestion
  TCreationAttributes extends {} = TAttributes
```

**Review by ephys — CHANGES_REQUESTED** [review]
Other than that, if tests pass (and they do), lgtm!

File: src/hooks.d.ts (hunk 1 above)
Comment (bnorbi95) [reply src/hooks.d.ts]: Fixed!

**Review by ephys — APPROVED** [review]

**Review by fzn0x — APPROVED** [review]

**Comment by ephys** [comment]
Failing tests are flaky tests, we can merge

**Comment by fzn0x** [comment]
Thank you for the initiatives!

**Pull request merged**